File: Organização de computador/Exercícios/Atividade02/inverArry.c
#include <stdio.h>
//DANIEL LIMA DE SOUZA

// Invete o arry
void invertArray(int *vetor){
    int aux;
    int i,j;
    
    for(i=0,j= 9; i < 5 ; i++,j--){
        aux = vetor[i];
        vetor[i] = vetor[j];
        vetor[j] = aux;
    }
}

//Soma os elementos de índice par do arry
int sumArry(int *vetor){
    int soma = 0;
    for(int i = 0; i < 10 ;i++, vetor++){
        soma += (i % 2 == 0) ? (*vetor): 0;
    }
    return (soma);
}

//Mostra na tela os elementos no estado atual
void printArry(int *vetor){
    for(int i = 0;i < 10; i++) printf(" %d ", vetor[i]);
}

// Função principal
int main() {
    int arry[]= {1,2,3,4,5,6,7,8,9,10}; 
    invertArray(arry); 
    printArry(arry);
    printf("\n A soma dos elementos de índice par é = %i\n", sumArry(arry));
    return 0;
}

File: COMPUTER ORGANIZATION/Exercícios/Atividade 02/inverArry.java
public class inverArry {

    //Função principal
    public static void main(String[] args) {
        int vetor[] = {1,2,3,4,5,6,7,8,9,10};
        invertVetor(vetor);
        System.out.println("Vetor invertido:");
        for(int valuer : vetor) System.out.printf("%d ",valuer);
        System.out.printf("\nA soma dos elementos de índice par é = [%d]\n",SumArry(vetor));
        
    }

    //Inverte os elementos do vetor
    public static void invertVetor(int vetor[]){
        int aux,i,j;
        for (i = 0, j = 9; i < 5; i++, j--){
            aux = vetor[i];
            vetor[i] = vetor[j];
            vetor[j] = aux;
        }
    }

    //Soma os elementos de índice par do vetor
    public static int SumArry(int vetor[]) {
        int soma=0;
        for(int i=0; i < 10; i++){
            soma += (i % 2 == 0) ? vetor[i] : 0;
        }
        return (soma);
    }
}